packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol

```diff
@@ -55,87 +84,125 @@ contract ZoraCreator1155PremintExecutorImpl is Ownable2StepUpgradeable, UUPSUpgr
     /// @param quantityToMint How many tokens to mint to the executor of this transaction once the token is created
-    /// @param mintComment A comment to associate with the mint action
+    /// @param mintArguments Abi encoded additional mint arguments: including mintComment and mintReferral
+    function premint(
+        ContractCreationConfig calldata contractConfig,
+        PremintConfigV2 calldata premintConfig,
+        bytes calldata signature,
+        uint256 quantityToMint,
+        bytes calldata mintArguments
+    ) external payable returns (uint256 newTokenId) {
+        (bytes memory encodedPremint, bytes32 premintVersion) = PremintEncoding.encodePremintV2(premintConfig);
+        address fixedPriceMinter = premintConfig.tokenConfig.fixedPriceMinter;
+        uint32 uid = premintConfig.uid;
+
+        // we wrap this here to get around stack too deep issues
+        {
+            newTokenId = _premint({
+                contractConfig: contractConfig,
+                encodedPremintConfig: encodedPremint,
+                premintVersion: premintVersion,
+                signature: signature,
+                quantityToMint: quantityToMint,
+                fixedPriceMinter: fixedPriceMinter,
+                uid: uid,
+                mintArguments: mintArguments
+            });
+        }
+    }
+
+    /// Creates a new token on the given erc1155 contract on behalf of a creator, and mints x tokens to the executor of this transaction.
+    /// If the erc1155 contract hasn't been created yet, it will be created with the given config within this same transaction.
+    /// The creator must sign the intent to create the token, and must have mint new token permission on the erc1155 contract,
+    /// or match the contract admin on the contract creation config if the contract hasn't been created yet.
+    /// Contract address of the created contract is deterministically generated from the contract config and this contract's address.
+    /// @param contractConfig Parameters for creating a new contract, if one doesn't exist yet.  Used to resolve the deterministic contract address.
+    /// @param premintConfig Parameters for creating the token, and minting the initial x tokens to the executor.
+    /// @param signature Signature of the creator of the token, which must match the signer of the premint config, or have permission to create new tokens on the erc1155 contract if it's already been created
+    /// @param quantityToMint How many tokens to mint to the executor of this transaction once the token is created
+    /// @param mintArguments Abi encoded additional mint arguments: including mintComment and mintReferral
     function premint(
         ContractCreationConfig calldata contractConfig,
         PremintConfig calldata premintConfig,
         bytes calldata signature,
         uint256 quantityToMint,
-        string calldata mintComment
+        bytes memory mintArguments
```

Comment:
todo:

* make mintArguments the same as `mintArguments` on 1155 (i.e. being able to specify mint recipient)
* mintReferral is it's own argument